website/api.php

```diff
@@ -482,22 +481,24 @@ function doWB($bid,$uid,$damage,$wbhp,$isSuper) {
                     ++$ids[array_search($row1["mid"],$WB)];
                     ++$modes[$row1["mode"]];
                 }
-                $forceWB = false;
+                $forceWB = false; // swb
                 $forceNH = false;
                 $forceH = false;
                 if ($modes[2]+$modes[3]<=5) $forceWB = true;
                 if ($modes[0]+$modes[2]<=20) $forceNH = true;
                 if ($modes[1]+$modes[3]<=20) $forceH = true;
                 $now=time();
-                $wb = $WB[rand(0,count($WB)-1)];
-                for ($i=0; $i<count($ids); ++$i) {
-                    if ($ids[$i]<7) $wb=$WB[$i];
-                }
+				do { // if we need a NH, prevent bornag from spawning
+					$wb = $WB[rand(0,count($WB)-1)];
+					for ($i=0; $i<count($ids); ++$i) {
+						if ($ids[$i]<7) $wb=$WB[$i];
+					}
+				} while (!$forceNH || $wb==186);
```

Comment:
This suspiciosly looks like a neverending loop unless there is a forced NH. I think it should be
```suggestion
				} while ($forceNH && $wb==186);
```

Reply:
Doh, I wanted to write || $wb!=186, but ofc your code is more readable, thx.